addons/web_editor/static/src/js/widgets.js

```diff
@@ -775,85 +775,83 @@ var fontIconsDialog = Widget.extend({
-    var youkuMatch = url.match(youkuRegExp);
+    var ykuRegExp = /(.*).youku\.com\/(v_show\/id_|embed\/)(.+)/;
+    var ykuMatch = url.match(ykuRegExp);
+
+    var $video = $('<iframe>').width(1280).height(720).attr('frameborder', 0).addClass('o_video_dialog_iframe');
+    var $alert = $('<div/>').addClass('alert o_video_dialog_iframe');
+    var video_type = 'yt';
+
+    if(!/^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i.test(url)){
+        $alert.addClass('alert-danger').text(_t('The provided url it\'s not valid' ));
+        return {$content: $alert, success: false};
+    }
+
+    var autoplay = options.autoplay ? "?autoplay=1" : "?autoplay=0";
 
-    var $video = $('<iframe>');
     if (ytMatch && ytMatch[1].length === 11) {
-        var youtubeId = ytMatch[1];
-        $video = $('<iframe>', {
-            src: '//www.youtube.com/embed/' + youtubeId,
-            width: '640',
-            height: '360'
-        });
-    } else if (igMatch && igMatch[0].length) {
-        $video = $('<iframe>', {
-            src: igMatch[0] + '/embed/',
-            width: '612',
-            height: '710',
-            scrolling: 'no',
-            allowtransparency: 'true'
-        });
-    } else if (vMatch && vMatch[0].length) {
-        $video = $('<iframe>', {
-            src: vMatch[0] + '/embed/simple',
-            width: '600',
-            height: '600',
-            class: 'vine-embed'
-        });
+        $video.attr('src', '//www.youtube.com/embed/' + ytMatch[1] + autoplay);
+    } else if (insMatch && insMatch[2].length) {
+        $video.attr('src', '//www.instagram.com/p/' + insMatch[2] + '/embed/');
+        video_type = 'ins';
+    } else if (vinMatch && vinMatch[0].length) {
+        $video.attr('src', vinMatch[0] + '/embed/simple');
+        video_type = 'vin';
     } else if (vimMatch && vimMatch[3].length) {
-        $video = $('<iframe>', {
-            src: '//player.vimeo.com/video/' + vimMatch[3],
-            width: '640',
-            height: '360'
-        });
+        $video.attr('src', '//player.vimeo.com/video/' + vimMatch[3] + autoplay);
+        video_type = 'vim';
     } else if (dmMatch && dmMatch[2].length) {
-        $video = $('<iframe>', {
-            src: '//www.dailymotion.com/embed/video/' + dmMatch[2],
-            width: '640',
-            height: '360'
-        });
-    } else if (youkuMatch && youkuMatch[1].length) {
-        $video = $('<iframe>', {
-            height: '498',
-            width: '510',
-            src: '//player.youku.com/embed/' + youkuMatch[1]
-        });
+        var just_id = dmMatch[2].replace('video/','');
+        $video.attr('src', '//www.dailymotion.com/embed/video/' + just_id + autoplay);
+        video_type = 'dm';
+    } else if (ykuMatch && ykuMatch[3].length) {
+        var yku_id = ~ykuMatch[3].indexOf( ".html?")? ykuMatch[3].substring(0, ykuMatch[3].indexOf( ".html?")) : ykuMatch[3];
```

Comment:
Well, I did not know what the `~` operator was but I now know that we should not use it :p 
=>
```javascript
ykuMatch[3].indexOf('.html?') >= 0 ? ...
```